engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java

```diff
@@ -739,13 +745,21 @@ public static InetSocketAddress getRMWebAddress(Configuration conf, String rmId)
 
   public static InetSocketAddress getRMWebAddress(Configuration conf, boolean sslEnabled, String rmId)
   {
-    rmId = (rmId == null) ? "" : ("." + rmId);
+    boolean isHA = (rmId != null);
+    rmId = isHA ? ("." + rmId) : "";
```

Comment:
Should modification to `rmId` be done after the call to `conf.set(ConfigUtils.RM_HA_ID, rmId)`? It is not modified in `getRMHAAddress()`.